server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -75,17 +83,33 @@ private RolloverResult(String rolloverIndexName, String sourceIndexName, Cluster
         }
     }
 
-    public RolloverResult rolloverClusterState(ClusterState currentState, String aliasName, String newIndexName,
+    public RolloverResult rolloverClusterState(ClusterState currentState, String aliasOrDataStream, String newIndexName,
                                                CreateIndexRequest createIndexRequest, List<Condition<?>> metConditions,
                                                boolean silent) throws Exception {
+        validate(currentState.metadata(), aliasOrDataStream, newIndexName);
```

Comment:
I wonder if for data streams we should disallow applying aliases, mappings and settings to the new index? It feels odd that you cannot specify them on create data stream but can on rollover?